packages/mcl/src/src/mcl/commands/compare_disko.d

```diff
@@ -0,0 +1,188 @@
+module mcl.commands.compare_disko;
+
+import mcl.utils.env : optional, parseEnv;
+import mcl.utils.nix : nix;
+import mcl.utils.path : getTopLevel;
+import mcl.utils.process : execute;
+import mcl.utils.log : errorAndExit;
+
+import std.json : JSONValue;
+import std.file : write;
+import std.logger : infof;
+import std.algorithm : uniq;
+import std.array;
+import std.meta : Filter;
+
+struct Params
+{
+    @optional() string baseBranch;
+
+    void setup(){
+        if (baseBranch == null){
+            baseBranch = "main";
+        }
+    }
+}
+
+enum ChangeStatus{
+    Unchanged,
+    Changed,
+    Removed,
+    New
+}
+
+struct MachineChanges{
+    string machine;
+    ChangeStatus _config;
+    ChangeStatus _create;
+}
+
+enum string[] diskoOptions = [ __traits(allMembers, MachineChanges)[1 .. $] ];
+
+string statusToSymbol(ChangeStatus s) {
+    final switch(s){
+        case ChangeStatus.Unchanged:
+            return "🟩";
+            break;
+        case ChangeStatus.Changed:
+            return "⚡";
+            break;
+        case ChangeStatus.Removed:
+            return "🗑";
+            break;
+        case ChangeStatus.New:
+            return "🧩";
+            break;
+    }
+}
+
+// TODO: handle case where a machine is missing from one branch
+export void compare_disko(string[] args){
+    const params = parseEnv!Params;
+
+    JSONValue configurations = nix.flake!JSONValue("", [], "show");
+
+    string[] machinesNew;
+    foreach (string k, JSONValue v; configurations["nixosConfigurations"]){
+        if (k[$-3 .. $] != "-vm"            &&
+            k != "gitlab-runner-container"  &&
+            k != "minimal-container"        )
+        {
+            machinesNew ~= k;
+        }
+    }
```

Comment:
Consider using appender!string instead of "~", as it's faster and more efficient.
https://forum.dlang.org/thread/ppbczcbqhaqalgwcjbba@forum.dlang.org